catalyst/utils/components.py

```diff
@@ -19,14 +26,69 @@
 from catalyst.utils.torch import get_device
 
 
+def _patch_forward(model):
+    import apex
+
+    input_caster_lambda = (
+        lambda tensor: tensor.to(
+            apex.amp._amp_state.opt_properties.options["cast_model_type"]
+        )
+        if tensor.is_floating_point()
+        else tensor
+    )
+    output_caster_lambda = (
+        lambda tensor: tensor.to(
+            apex.amp._amp_state.opt_properties.options.get(
+                "cast_model_outputs", torch.float32
+            )
+        )
+        if tensor.is_floating_point()
+        else tensor
+    )
+
+    model.forward = lambda *args, old_fwd=model.forward, input_caster=input_caster_lambda, output_caster=output_caster_lambda, **kwargs: apex.amp._initialize.applier(
+        old_fwd(
+            *apex.amp._initialize.applier(args, input_caster),
+            **apex.amp._initialize.applier(kwargs, input_caster),
+        ),
+        output_caster,
+    )
+    return model
+
+
+# apex issue https://github.com/deepset-ai/FARM/issues/210
+# solution: https://github.com/NVIDIA/apex/issues/503#issuecomment-566181771
+def _wrap_into_data_parallel_with_apex(
+    model: RunnerModel, optimizer: Optimizer, distributed_params: Dict
+):
+    if isinstance(model, nn.Module):
+        model = nn.Sequential(model)
```

Comment:
why are we checking the model for ``nn.Module``, but cast it to ``nn.Sequential`` after that?